### ISSUE TITLE
Support `Java 21` LTS

### ISSUE THREAD

**Issue by dongjoon-hyun** [apache/orc#1498]
Java 21 LTS GA is scheduled on 2023 September
- https://openjdk.org/projects/jdk/21/

**Comment by deshanxiao** [comment]
JDK 21 is available on September 19th, could we start work on this part?

**Comment by dongjoon-hyun** [comment]
Thank you for having interests in this domain.

Until now, we did the following and I'm waiting for Eclipse Temurin Java 21 release in order to upgrade GitHub Action job from `21-ea` to the official `21` GA version.
```
ORC-1443: Enforce Java version
ORC-1426: Use Java `21-ea` instead of `20` in GitHub Action
```

Do you have something to do in your mind, @deshanxiao ?

**Comment by dongjoon-hyun** [comment]
If you have something, feel free to start to create PRs. You're welcome. :)

**Comment by dongjoon-hyun** [comment]
I checked JDK distributions Today once more and created https://github.com/apache/orc/pull/1620 .

**Comment by dongjoon-hyun** [comment]
This is resolved via #1620
